### ISSUE TITLE
Unit trying to leave city blocked

### ISSUE THREAD

**Issue by fire-eggs** [fire-eggs/CivOne#123]
Seems like an invalid constraint.

```
OOLL
LCLB
LELL
```
Trying to move unit from my city C, with enemy unit E, barbarian unit B.
Could not move unit left, right, or down-and-left, thanks to E. Could not
move up-and-right thanks to B.

**Comment by iegik** [comment]
> When leaving a city a unit should be able to move into any Land spot
Yes, in such situation land unit from city can move to any land direction. If it is an attack unit - it can attack, if it is settler - it can move only on free space or pass the turn. If it is sea unit - it can attack and move to the ocean.

**Comment by iegik** [comment]
You can use Diplomat to move other units around enemy's cities.
1. Put Diplomat first on enemy's territory
2. Move other unit to the place there Diplomat stays

This way you can capture enemy's territory

**Comment by iegik** [comment]
<img width="419" alt="Screenshot 2021-09-19 at 21 22 07" src="https://user-images.githubusercontent.com/175293/133938462-75671eba-3af3-4d6a-b858-c73d4043a793.png">

**Comment by fire-eggs** [comment]
Problem must be in BaseUnit.cs, `CanMoveTo()`. 

Comment from code, bolding added:
// Cannot move from a square adjacent to enemy unit to a square adjacent to enemy unit
// but _can_ move to square occupied by own units **or to any undefended city square**

The current code doesn't include the bolded condition in this scenario. Something along the lines of:
```
Map[X,Y].HasCity && !moveTarget.Units.Any(u=>u.Owner != Owner)
```
needs to be added as an "OK to move" test.
